WireDomain/Sources/WireDomain/Notifications/Components/EnvironmentComponent.swift

```diff
@@ -0,0 +1,164 @@
+    }
+
+    var appMainBundle: Bundle {
+        let mainBundle: Bundle
+
+        let runningInExtension = Bundle.main.bundlePath.hasSuffix(".appex")
+
+        if runningInExtension {
+            let extensionBundleURL = Bundle.main.bundleURL
+            let mainAppBundleURL = extensionBundleURL.deletingLastPathComponent().deletingLastPathComponent()
+            guard let bundle = Bundle(url: mainAppBundleURL) else { fatalError("Failed to find main app bundle") }
+            mainBundle = bundle
+        } else {
+            mainBundle = .main
+        }
+        return mainBundle
+    }
+
+    var proxySettings: ProxySettings? {
+        get async {
+            guard let proxy = legacyBackendEnvironment.proxy else { return nil }
+
+            let keychain = WireFoundation.Keychain()
+            let usernameItemID = "proxy-\(proxy.host):\(proxy.port)-username"
+            let passwordItemID = "proxy-\(proxy.host):\(proxy.port)-password"
+
+            let proxyUsername: String? = try? await keychain.fetchItem(
+                query: [
+                    .itemClass(.genericPassword),
+                    .account(usernameItemID),
+                    .returningData(true)
+                ]
+            )
+
+            let proxyPassword: String? = try? await keychain.fetchItem(
+                query: [
+                    .itemClass(.genericPassword),
+                    .account(passwordItemID),
+                    .returningData(true)
+                ]
+            )
+
+            if proxy.needsAuthentication {
+                guard let proxyUsername, let proxyPassword else {
+                    fatalInternal(
+                        "Proxy needs authentication but credentials are missing"
+                    )
+
+                    return nil
+                }
+
+                return .authenticated(
+                    host: proxy.host,
+                    port: proxy.port,
+                    username: proxyUsername,
+                    password: proxyPassword
+                )
+            } else {
+                return .unauthenticated(
+                    host: proxy.host,
+                    port: proxy.port
+                )
+            }
+        }
+    }
+
+    // MARK: - Private
+
+    private var legacyBackendEnvironment: WireDataModel.BackendEnvironment {
+        guard let backendEnvironmentTypeOverride else {
+            fatalError()
+        }
+
+        let environmentType = EnvironmentType(
+            stringValue: backendEnvironmentTypeOverride
+        )
+
+        guard let backendEnvironment = BackendEnvironment(
+            userDefaults: userDefaults,
+            configurationBundle: backendBundle,
+            environmentType: environmentType
+        ) else {
+            fatalError("Malformed backend configuration data")
+        }
+
+        return backendEnvironment
+    }
+
+    private var userDefaults: UserDefaults {
+        let userDefaults = UserDefaults.standard
```

Comment:
isn't it UserDefaults.applicationGroup instead, otherwise how does NSE knows the value of main Application backend